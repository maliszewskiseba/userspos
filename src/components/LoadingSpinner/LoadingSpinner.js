import React from "react";
import styled, { keyframes } from "styled-components";
import { rem } from "polished";

import i18n from "../../i18n";
import mainTheme from "../../helpers/mainTheme";

const LoadingSpinnerBox = styled.div`
  width: 100%;
  height: 98vh;
  display: flex;
  justify-content: center;
  align-items: center;
  position: absolute;
  top: 0;
  left: 0;
`;

const LoadingSpinnerText = styled.span`
  position: absolute;
  display: inline-block;
  font-family: ${({ theme }) => theme.font.family};
  font-weight: ${({ theme }) => theme.font.weightBold};
  font-size: ${({ theme }) => theme.font.sizeBig};
  color: rgba(0, 0, 0, 0.6);
`;
LoadingSpinnerText.defaultProps = { theme: mainTheme };

const rotate = keyframes`
  from {
    transform: rotate(0deg);
    opacity: 0.2;
    border-top: ${rem(2)} dashed red;
    border-bottom: ${rem(2)} dashed red;
  }

  to {
    opacity:1
    transform: rotate(360deg);
    border-top: ${rem(6)} solid blue;
    border-bottom: ${rem(6)} solid blue;
  }
`;

const LoadingSpinnerInnerCircle = styled.div`
  width: ${rem(150)};
  height: ${rem(150)};
  border-radius: 50%;
  border-top: ${rem(3)} solid black;
  border-bottom: ${rem(3)} solid black;
  animation: ${rotate} 2s ease-in-out infinite;
  opacity: 0.2;
`;

const LoadingSpinner = () => (
  <LoadingSpinnerBox>
    <LoadingSpinnerInnerCircle />
    <LoadingSpinnerText>{i18n.t("custom.loading")}</LoadingSpinnerText>
  </LoadingSpinnerBox>
);

export default LoadingSpinner;
