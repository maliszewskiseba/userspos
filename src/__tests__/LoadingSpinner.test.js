import React from 'react';
import { create } from 'react-test-renderer';
import LoadingSpinner from '../components/LoadingSpinner/LoadingSpinner';

describe('Loading spinner component', () => {
  test('it matches the snapshot', () => {
    const component = create(<LoadingSpinner />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
