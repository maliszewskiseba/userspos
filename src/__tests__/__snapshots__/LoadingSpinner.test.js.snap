// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading spinner component it matches the snapshot 1`] = `
<div
  className="sc-bdVaJa eczqmb"
>
  <div
    className="sc-htpNat dGFVaf"
  />
  <span
    className="sc-bwzfXH cSFKPS"
  >
    Loading
  </span>
</div>
`;
